python/tank_vendor/shotgun_authentication/session_cache.py

```diff
@@ -67,81 +66,242 @@ def _get_local_site_cache_location(base_url):
         root = os.path.join(os.environ["APPDATA"], "Shotgun")
     elif sys.platform.startswith("linux"):
         root = os.path.expanduser("~/.shotgun")
+    return root
+
+
+def _get_local_site_cache_location(base_url):
+    """
+    Returns the location of the site cache root based on a site.
+
+    :param base_url: Site we need to compute the root path for.
+
+    :returns: An absolute path to the site cache root.
+    """
+    return os.path.join(
+        _get_cache_location(),
+        # get site only; https://www.foo.com:8080 -> www.foo.com
+        urlparse.urlparse(base_url)[1].split(":")[0]
+    )
 
-    # get site only; https://www.foo.com:8080 -> www.foo.com
-    base_url = urlparse.urlparse(base_url)[1].split(":")[0]
 
-    return os.path.join(root, base_url)
+def _get_authentication_cache_location(base_url):
+    """
+    Returns the location for the authentication cache folder for a given site.
+
+    :param base_url: Site we need to compute the authentication cache path for.
+
+    :returns: An absolute path to the authentication cache root.
+    """
+    return os.path.join(
+        _get_local_site_cache_location(base_url),
+        "authentication"
+    )
 
 
-def _get_session_data_location(base_url):
+def _get_users_file_location(base_url):
     """
-    Returns the location of the session file on disk for a specific site.
+    Returns the location of the users file on disk for a specific site.
 
     :param base_url: The site we want the login information for.
 
     :returns: Path to the login information.
     """
     return os.path.join(
-        _get_local_site_cache_location(base_url),
+        _get_authentication_cache_location(base_url),
+        "users.yml"
+    )
+
+
+def _get_current_user_file_location(base_url):
+    """
+    Returns the location of the users file on disk for a specific site.
+
+    :param base_url: The site we want the login information for.
+
+    :returns: Path to the login information.
+    """
+    return os.path.join(
+        _get_authentication_cache_location(base_url),
+        "current_user.yml"
```

Comment:
So to add to my other comment below about the site wide config file - I think if we name this file `CACHE/SITE_NAME/authentication.yml` it will be consistent with the site wide config file. I think would be nice and consistent.